### PR TITLE
chore: Improve development experience about global variables

### PR DESCRIPTION
**Pull request by liuxingbaoyu**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | `Fixes #1, Fixes #2` 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

1. I added global declarations for `USE_ESM, IS_STANDALONE, PACKAGE_JSON, BABEL_8_BREAKING`, now we don't need to declare them in every file.
2. I created a special package that can only be used for development within the repository.
It is very convenient for us to use `USE_ESM` and `require, __dirname` etc.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/15646"></a>

### CODE REVIEW

File: scripts/repo-utils/index.cjs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    path.join(__dirname, "../../.module-type"),
+    "utf-8"
+  ).trim();
+  USE_ESM = type === "module";
+} catch (_) {}
+
+exports.USE_ESM = USE_ESM;
+
+if (typeof jest !== "undefined") {
+  exports.itNoESM = USE_ESM ? it.skip : it;
```
Comment (nicolo-ribaudo) [inline scripts/repo-utils/index.cjs:21]: Are these imported anywhere?

File: scripts/repo-utils/index.cjs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  };
+  exports.itGteNoESM = function (version) {
+    return !USE_ESM && semver.gte(process.version, version) ? it : it.skip;
+  };
+  exports.itGte = function (version) {
+    return semver.gte(process.version, version) ? it : it.skip;
+  };
+}
+
+exports.globals = function (metaUrl) {
```
Comment (nicolo-ribaudo) [inline scripts/repo-utils/index.cjs:34]: I'd prefer to call this `commonJS`, since they are not globals :)

Comment (liuxingbaoyu) [reply scripts/repo-utils/index.cjs]: Sounds good! I'm a mess when it comes to naming things. :)

File: scripts/generators/tsconfig.js [hunk 3]
```diff
@@ -103,7 +103,13 @@ fs.writeFileSync(
...
         include: tsPkgs
           .map(({ relative }) => `${relative}/src/**/*.ts`)
-          .concat(["./packages/babel-parser/typings/*.d.ts"]),
+          .concat([
+            "./lib/globals.d.ts",
+
+            "./scripts/repo-utils/*.d.ts",
+
+            "./packages/babel-parser/typings/*.d.ts",
+          ]),
```
Comment (nicolo-ribaudo) [inline scripts/generators/tsconfig.js:112]:
```suggestion
          .concat([
            "./lib/globals.d.ts",
            "./scripts/repo-utils/*.d.ts",
            "./packages/babel-parser/typings/*.d.ts",
          ]),
```

**Review by nicolo-ribaudo — APPROVED** [review]
This looks good :)

If we accidentally add $repo-utils as a dependency in a `src/` folder, do we get a linting (or TS) error?

**Comment by liuxingbaoyu** [comment]
Good idea! I tried it and the linter would report `import/no-extraneous-dependencies`.

**Review by JLHwung — APPROVED** [review]

**Pull request merged**